src/cid.ts

```diff
@@ -177,59 +139,42 @@ export class CID {
     }
   }
 
-  /**
-   * @param {unknown} other
-   * @returns {other is CID<Data, Format, Alg, Version>}
-   */
-  equals (other) {
+  equals (other: unknown): other is CID<Data, Format, Alg, Version> {
     return CID.equals(this, other)
   }
 
-  /**
-   * @template {unknown} Data
-   * @template {number} Format
-   * @template {number} Alg
-   * @template {API.Version} Version
-   * @param {API.Link<Data, Format, Alg, Version>} self
-   * @param {unknown} other
-   * @returns {other is CID}
-   */
-  static equals (self, other) {
+  static equals <Data, Format extends number, Alg extends number, Version extends API.Version>(self: API.Link<Data, Format, Alg, Version>, other: unknown): other is CID {
     const unknown =
-      /** @type {{code?:unknown, version?:unknown, multihash?:unknown}} */ (
-        other
+      (
+        other as { code?: unknown, version?: unknown, multihash?: unknown }
       )
```

Comment:
```suggestion
    const unknown = other as { code?: unknown, version?: unknown, multihash?: unknown }
```

I assume the `(` is unnecessary here?

Reply:
I'll go through and delete extraneous parens.